compiler/backend/src/org/jetbrains/kotlin/codegen/FunctionCodegen.java

```diff
@@ -647,7 +647,20 @@ else if (isCompatibilityStubInDefaultImpls(functionDescriptor, context, jvmDefau
                     KtDeclarationWithBody declaration = ((ClosureGenerationStrategy) strategy).getDeclaration();
                     BindingContext bindingContext = typeMapper.getBindingContext();
                     if (declaration instanceof KtFunctionLiteral && isLambdaPassedToInlineOnly((KtFunction) declaration, bindingContext)) {
-                        lineNumber = LOCAL_VARIABLE_INLINE_ARGUMENT_SYNTHETIC_LINE_NUMBER;
+                        // We want the debugger to be able to break both on the inline call itself, plus on each
+                        // invocation of the inline lambda passed to it. For that to happen there needs to be at least
+                        // one different line number in between those breakpoints for the VM to emit a locatable event.
+                        // @InlineOnly functions, however, contain no line numbers, so if the lambda is single-line,
+                        // the entire call will "meld" into a single region. To break it up, we insert a different line
+                        // number that is remapped by the SMAP back to the lambda.
+                        Integer lambdaLine = CodegenUtil.getLineNumberForElement(declaration, false);
+                        Integer firstBodyLine = CodegenUtil.getLineNumberForElement(declaration.getBodyExpression(), false);
```

Comment:
Please note that 'declaration.getBodyExpression()' is Nullable

Could you also add test for next cases to test absent of additional mapping:
// IGNORE_BACKEND_MULTI_MODULE: JVM_IR
// FILE: 1.kt
package test

@Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE")
@kotlin.internal.InlineOnly
inline fun <T, R> T.myLet(block: (T) -> R) = block(this)

// FILE: 2.kt
import test.*

fun box(): String {
    // in order to let the debugger break on it (KT-23064)
    return "O".myLet { param -> 
            it + "K" 
    }
}
and 
    return "O".myLet { fun foo(a: String) {} -> 
         a + "K" 
    }

Reply:
Am I wrong in assuming that function literals should never have a null body?

Reply:
You are right, but there could be some suppressions. It best to keep some kind of robustness with explicit checks, e.g. see FunctionGenerationStrategy.FunctionDefault

Reply:
Added both tests. Thanks for pointing out the anonymous `fun` case - looks like it was still broken because the checks around the code were too restrictive. Unfortunately, I can't figure a way to detect if a `KtNamedFunction` is single-line - do you have any pointers about this?

Reply:
I just realized that checking for single-line functions is not correct anyway: even it they are not single-line, we still need to insert a different line number *in between two calls*. Consider this test:
```
fun main(args: Array<String>) {
    1.alsoTwice {
        println(it)
    }
}

@Suppress("INVISIBLE_MEMBER", "INVISIBLE_REFERENCE")
@kotlin.internal.InlineOnly
inline fun <T> T.alsoTwice(block: (T) -> Unit) {
    block(this)
    block(this)
}
```
If a breakpoint is placed on `println(it)` it will only be triggered once because there is no different line number in between two `println` invocations.

Reply:
As I remember `inlineOnly` functions in most cases are quite simple wrappers around java ones. But it's very intresting notice cc @yanex

Reply:
Looks like this is actually broken even without `InlineOnly`, and adding a line number does not help.